rust/crd/src/lib.rs

```diff
@@ -315,7 +327,8 @@ impl HbaseRole {
             resources,
             logging: product_logging::spec::default_logging(),
             affinity: get_affinity(cluster_name, self, hdfs_discovery_cm_name),
-            graceful_shutdown_timeout: Some(graceful_shutdown_timeout),
+            graceful_shutdown_timeout,
+            requested_secret_lifetime,
```

Comment:
nit (optional): I would suggest using the Some() here (instead of above), so that it is enforced that every possible code path produces a valid non-default value. But if you have stronger feelings, please ignore :sweat_smile: 
```suggestion
            graceful_shutdown_timeout: Some(graceful_shutdown_timeout),
            requested_secret_lifetime: Some(requested_secret_lifetime),
```

Reply:
I don't have strong feelings but I don't understand. What code path can lead to an invalid value?

Reply:
If we do it like this
```diff
diff --git a/rust/crd/src/lib.rs b/rust/crd/src/lib.rs
index e9930e7..321b1d3 100644
--- a/rust/crd/src/lib.rs
+++ b/rust/crd/src/lib.rs
@@ -310,15 +310,15 @@ impl HbaseRole {
         };
 
         let graceful_shutdown_timeout = match &self {
-            HbaseRole::Master => Some(Self::DEFAULT_MASTER_GRACEFUL_SHUTDOWN_TIMEOUT),
-            HbaseRole::RegionServer => Some(Self::DEFAULT_REGION_SERVER_GRACEFUL_SHUTDOWN_TIMEOUT),
-            HbaseRole::RestServer => Some(Self::DEFAULT_REST_SERVER_GRACEFUL_SHUTDOWN_TIMEOUT),
+            HbaseRole::Master => Self::DEFAULT_MASTER_GRACEFUL_SHUTDOWN_TIMEOUT,
+            HbaseRole::RegionServer => Self::DEFAULT_REGION_SERVER_GRACEFUL_SHUTDOWN_TIMEOUT,
+            HbaseRole::RestServer => Self::DEFAULT_REST_SERVER_GRACEFUL_SHUTDOWN_TIMEOUT,
         };
 
         let requested_secret_lifetime = match &self {
-            HbaseRole::Master => Some(Self::DEFAULT_MASTER_SECRET_LIFETIME),
-            HbaseRole::RegionServer => Some(Self::DEFAULT_REGION_SECRET_LIFETIME),
-            HbaseRole::RestServer => Some(Self::DEFAULT_REST_SECRET_LIFETIME),
+            HbaseRole::Master => Self::DEFAULT_MASTER_SECRET_LIFETIME,
+            HbaseRole::RegionServer => Self::DEFAULT_REGION_SECRET_LIFETIME,
+            HbaseRole::RestServer => Self::DEFAULT_REST_SECRET_LIFETIME,
         };
 
         HbaseConfigFragment {
@@ -327,8 +327,8 @@ impl HbaseRole {
             resources,
             logging: product_logging::spec::default_logging(),
             affinity: get_affinity(cluster_name, self, hdfs_discovery_cm_name),
-            graceful_shutdown_timeout,
-            requested_secret_lifetime,
+            graceful_shutdown_timeout: Some(graceful_shutdown_timeout),
+            requested_secret_lifetime: Some(requested_secret_lifetime),
         }
     }
```
You can directly tell `graceful_shutdown_timeout` and `requested_secret_lifetime` are always set as a default value, no matter what.
With the current state you would need to look at the code above to be sure that it is always set, and operators can rely on the value always being `Some`.

But it's a nit

Reply:
[applied](https://github.com/stackabletech/hbase-operator/pull/598/commits/8c4387ae85fd5c296c1edfe734c5635d33f7dd68)

Reply:
Thanks!